network/gossipNode.go

```diff
@@ -80,8 +80,10 @@ type GossipNode interface {
 	// ClearHandlers deregisters all the existing message handlers.
 	ClearHandlers()
 
-	// RegisterProcessors adds to the set of given message processors.
-	RegisterProcessors(dispatch []TaggedMessageProcessor)
+	// RegisterValidatorHandlers adds to the set of given message validation handlers.
+	// A difference with regular handlers is validation ones perform synchronous validation.
+	// Currently used as p2p pubsub topic validators.
+	RegisterValidatorHandlers(dispatch []TaggedMessageValidatorHandler)
```

Comment:
I guess we can completely remove `RegisterValidatorHandlers` and reuse `RegisterProcessors` instead? some refactoring for hybrid mode will be needed to support different handlers under the same `RegisterProcessors` api tho.